File: packages/axon-core-py/src/axon_core/lot.py
from __future__ import annotations
from typing import Literal, Optional
from pydantic import BaseModel
from .material import Quantity

LotStatus = Literal[
    "pending-iqc", "iqc-in-progress", "released",
    "quarantined", "rejected", "expired",
]


class AttachmentRef(BaseModel):
    id: str
    fileName: str
    mediaType: str
    size: Optional[int] = None
    uri: Optional[str] = None


class MaterialLot(BaseModel):
    id: str
    materialId: str
    supplierId: str
    supplierLotRef: Optional[str] = None
    quantity: Quantity
    receivedAt: str
    status: LotStatus
    expiresAt: Optional[str] = None
    storageLocation: Optional[str] = None
    coaId: Optional[str] = None

File: packages/axon-core-ts/src/lot.ts
import type { Quantity } from "./material.js";

export type LotStatus =
  | "pending-iqc"
  | "iqc-in-progress"
  | "released"
  | "quarantined"
  | "rejected"
  | "expired";

/** A receipt of material from a supplier, tracked through IQC and inventory. */
export interface MaterialLot {
  id: string;
  materialId: string;
  supplierId: string;
  supplierLotRef?: string;
  quantity: Quantity;
  receivedAt: string; // ISO-8601
  status: LotStatus;
  expiresAt?: string;
  storageLocation?: string;
  /** ID of the latest CoA attached to this lot. */
  coaId?: string;
}

export interface AttachmentRef {
  id: string;
  fileName: string;
  mediaType: string;
  size?: number;
  uri?: string;
}
